usage/src/com/cloud/usage/UsageSanityChecker.java

```diff
@@ -72,7 +72,9 @@ protected boolean checkItemCountByPstmt(CheckCase checkCase) throws SQLException
         try (PreparedStatement pstmt = conn.prepareStatement(checkCase.sqlTemplate)) {
             if(checkCase.checkId) {
                 pstmt.setInt(1, lastId);
-                pstmt.setInt(2, maxId);
+                if (maxId > 0) {
+                    pstmt.setInt(2, maxId);
```

Comment:
@nvazquez thanks for the explanation. Now I understood it a little bit more. These SLQs that are built using if(s)/else(s) and use few methods are a bit hard to follow sometimes without debugging it in runtime.

This was my only doubt. So, LGTM here.